packages/devtools_server/lib/src/handlers.dart

```diff
@@ -36,10 +36,50 @@ Future<shelf.Handler> defaultHandler() async {
 
   // Make a handler that delegates based on path.
   return (shelf.Request request) {
+    // The API handler takes all calls to api/.
+    if (Api.canHandle(request)) {
+      return Api.handle(request);
+    }
     return request.url.path.startsWith('packages/')
         // request.change here will strip the `packages` prefix from the path
         // so it's relative to packHandler's root.
         ? packHandler(request.change(path: 'packages'))
         : buildHandler(request);
   };
 }
+
+/// The DevTools server API.
+///
+/// This defines endpoints that serve all requests that come in over api/.
+class Api {
+  /// Determines whether or not [request] is an API call.
+  static bool canHandle(shelf.Request request) {
+    return request.url.path.startsWith('api/');
+  }
+
+  /// Handles all requests.
+  ///
+  /// To override an API call, pass in a subclass of [Api].
+  static FutureOr<shelf.Response> handle(shelf.Request request, [Api api]) {
+    api ??= Api();
+    switch (request.url.path) {
+      case 'api/logPageView':
+        return api.logPageView(request);
+      default:
+        return api.notImplemented;
+    }
+  }
+
+  /// Logs a page view in the DevTools server.
+  ///
+  /// In the open-source version of DevTools, Google Analytics handles this
+  /// without any need to involve the server.
+  FutureOr<shelf.Response> logPageView(shelf.Request request) async {
+    return notImplemented;
+  }
+
+  /// A [shelf.Response] for API calls that have not been implemented in this
+  /// server.
+  shelf.Response get notImplemented => shelf.Response.notFound(
```

Comment:
instead of a getter, can you make it a method and pass in the name of the unimplemented API?

```
shelf.Response notImplemented(String name) {
  return shelf.Response.notFound('$name API not implemented in this version of the DevTools server.');
}
```

Reply:
Great idea!